syntax = "proto2";
package oneflow;

import "oneflow/core/register/field_desc.proto";

message FieldHeaderDesc {
  optional FieldDescProto data_id = 1;
  optional FieldDescProto col_num = 2;
}

message BlobHeaderDescProto {
  required int32 max_col_num = 1;
  required int32 blob_mem_id = 2;
  required int32 instance_available_elem_cnt = 3;
  oneof header_type {
    FieldDescProto opaque_header = 4;
    FieldHeaderDesc field_header = 5;
  }
}

message BlobDescProto {
  required BlobHeaderDescProto header = 1;
  required FieldDescProto body = 2;
}
